front/src/applications/stdcm/utils/__tests__/simulationReportSheet.spec.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  generateCodeNumber,
+  formatCreationDate,
+  getStopDurationTime,
+  extractSpeedLimit,
+  getOperationalPointsWithTimes,
+} from 'applications/stdcm/utils/formatSimulationReportSheet';
+
+import { suggestedOperationalPoints, simulation, departure_time } from './sampleData';
+
+describe('Utils simulation report sheet functions', () => {
+  test('should return a formatted string', () => {
+    const codeNumber = generateCodeNumber();
+    expect(codeNumber).toMatch(/^\d{2}\d{2}-\d{3}-\d{3}$/);
+  });
+
+  test('should format the date correctly', () => {
+    const formattedDate = formatCreationDate(departure_time);
+    expect(formattedDate).toEqual({
+      day: '08',
+      month: '07',
+      year: 2024,
+      hours: 10,
+      minutes: 20,
+    });
+  });
+
+  test('should return correct time format', () => {
+    expect(getStopDurationTime(30)).toBe('30 sec');
+    expect(getStopDurationTime(120)).toBe('2 min');
+  });
+
+  test('should extract the speed limit correctly', () => {
+    const speedLimitByTag = 'Some tag - 50';
+    expect(extractSpeedLimit(speedLimitByTag)).toBe('50');
+  });
+
+  test('should return formatted operational points with times', () => {
+    const result = getOperationalPointsWithTimes(
```

Comment:
As this function is mainly base on multiple small utils doing the job, could you please these the small ones rather than this one ?

You could test : 
- addMinutesToTime > with normal time, when hour is midnight (btw how do you deal for english time ? if it's 06:32 pm)
- getStopDurationBetweenTwoPositions
- computeStopDepartureTime (by the way, I feel like the parameter names of this function are confusing. They should match the description of the function and if you need to specify their formats you can use this format : 
![image](https://github.com/user-attachments/assets/7a853090-1869-4d36-9552-ca2172c3cfe7)

Reply:
Fixed ! 
for the English version, we don't currently support the am/pm display on the application. This will be something to modify when it's the case.